First working draft of using http

Leidtier opened the pull request:
many adjustments for sending data over HTTP instead of Files

Please use this PR to test and commit changes

Leidtier committed:
First working draft of using http

YetAnotherModder committed:
Update README.md

Updated credits

YetAnotherModder committed:
Add holotape settings and misc changes

Add a selector for the HTTP port in the holotape
Add an an ON/OFF button for the microphone.
Correct port to 4999 on init
Correct the tooltip for the letter 'H' to 72 for the textinput hotkey in the holotape.
Move the microphone variable initialization to repository reinitializeVariables()

Leidtier committed:
Fixed the access problem to uninitialized arrays

YetAnotherModder committed:
Merge with MantellaMod 0.8.2

-Merge listener, conversation & repository with MantellaMod 0.8.2
-Add features from Mantella 0.8.2 in esp script: including perk, holotape rework
-Fix bug with stop conversation missing an unregister event for Pipboy close
-Add subtitles back in MantellaConversation script
-Fix crosshair function to initiate conversations
-Add GameEvent back in MantellaConversation Script

RevoluPowered commented:
Is this still being worked on? I am wondering if someone has ported the .pex to the latest fallout version or how hard this is to do? (can help out if required)

Leidtier commented:
It is, but Fallout 4 poses a couple of extra issues at the moment ;-)

YetAnotherModder committed:
Major commit - Multiple changes

Multiple scripts
 - Organized functions and events within labelled subsections for clarity
MantellaRepository
- Add in a timer with a check to do a conversation.CleanupConversation() when the End Conversastion holotape option is triggered if conversation.EndConversation() doesn't work
- Add notification when triggering options in the holotape for clarity
- Add an UnregisterForMenuOpenCloseEvent() when requesting to stop conversation to prevent it from triggering on any pipboy close event after being called.
-Add check to OnkeyDown to prevent keypress events from being called from within menus.
- Add a check for Crosshair actor to be filled before attempting to start a conversation with the actor.
 MantellaTargetListenerScript
-Remove MantellaTargetListenerScript - Functionality has been merged in MantellaEffectScript
MantellaListernerScript
- Move version checks in their own function
- Add an extra timer base check to call conversation.CleanupConversation() on game load if conversation.EndConversation() doesn't work
- Remove notifications when RadiantConversation fail to start.
- Change the ammokeyword filter so that it only filters out ammo being droppped because that's called when firing weapons.
- Rephrased the OnPlayerFireWeapon() to account for melee and unarmed attacks
MantellaEffectScript
- Remove property TargetRefAlias as it's no longer used
- Refactor OnEffectStart to give cleared feedback to the player
- Add OnEffectFinish event to deactivate event filters
- Add invetory and on hit filters to allow event tracking
- Add event tracking from MantellaTargetListenerScript
- Integrate a menu when adding a new NPC to the conversation to give the player the option to add the NPC or not
MantellaConversation
-Add MantellaSpell property as well as a dispel() call during Cleanup to manage game event listeners
-Add a call to generate screenshots right before sendRequestForPlayerInput() for upcoming vision analysis
-Add "Listening..." and "Thinking..." notifications for clearer user awareness of what the LLM is doing.
-Added a delayed call to CleanAllDictionary to prevent crashes if it's called in proximity of the the Stop() call.
-Add ResetEventSpamBlockers() on player response to allow the Listener Script to pick up on those again
-Add subtitles notification display (if allowed in the holotape, default is on)
- Add DispelAllMantellaMagicEffectsFromActors() to manage game events
- Add IsActorInConversation() function as a check that's called on EffectStart in MantellaEffectScript
MantellaConstants
	new KEY_CONTEXT_CUSTOMVALUES : mantella_vision_ready for custom vision analysis

Mantellaa.esp:
Add new holotape options for vision analysis
Remove the ReferenceAlias from MantellaEvents as they're not longer needed due to the listeners being move to MantellEffectSCript

YetAnotherModder committed:
Added a safety dispel scan

-Add a safety dispel scan to MantellaListenerScript on teleportEvent and on playerloadgame to remove any stragglers that may have MantellaEffect
-Add ScanCellForActors() to DispelAllMantellaMagicEffectsFromActors() to MantellaRepository

YetAnotherModder committed:
Misc updates and bug fixes

-Add a check for ongoing conversation to prevent a ctd from an overlapping Start() call with a Cleanupconversation() call
-Add an adjustable image resolution setting for image analysis to lower token cost and accelerate responses.
Moved a few notifications to trace analysis to prevent in game notification spam.
Mantellarepository : add a new checkAndUpdateVisionPipeline() to eventually allow for one shot image collection.
